packages/core/src/Modifiers.ts

```diff
@@ -162,14 +186,37 @@ export class Modifiers extends Array<Modifier> {
      * @param otherModifiers
      */
     areSameAs(otherModifiers: Modifiers): boolean {
-        const modifiersMap = new Map(this.map(a => [a, otherModifiers.find(b => a.isSameAs(b))]));
+        const modifiersMap = new Map(
+            this._contents.map(a => [a, otherModifiers.find(b => a.isSameAs(b))]),
+        );
         const aModifiers = Array.from(modifiersMap.keys());
         const bModifiers = Array.from(modifiersMap.values());
 
         const allAinB = aModifiers.every(a => a.isSameAs(modifiersMap.get(a)));
         const allBinA = otherModifiers.every(
-            b => bModifiers.includes(b) || b.isSameAs(this.get(b)),
+            b => bModifiers.includes(b) || b.isSameAs(this.find(b)),
         );
         return allAinB && allBinA;
     }
+    some(
+        callbackfn: (value: Modifier, index: number, array: Modifier[]) => unknown,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        thisArg?: any,
+    ): boolean {
+        return this._contents.some(callbackfn, thisArg);
+    }
+    every(
+        callbackfn: (value: Modifier, index: number, array: Modifier[]) => unknown,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        thisArg?: any,
+    ): boolean {
+        return this._contents.every(callbackfn, thisArg);
+    }
+    map<T>(
+        callbackfn: (value: Modifier, index: number, array: Modifier[]) => T,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
What ? Do we need the `thisArg` ? I mean, I never called any of these functions with a second argument ever and I didn't even know it existed...

Reply:
I figured if I'm going to make a proxy for a native method I should probably include all its arguments but if you think that's overkill I won't complain.

Reply:
I'm torn. I agree with you that it's better if the proxy perfectly matches the original signature. However, the signature is ugly in Typescript *and* is rather useless anyway. The `thisArg` is to specify the `this` used in the callback. But even without it, you can use `bind` on the callback function to set the `this` anyway, which is what I always did in such cases since I didn't know this parameter even existed.